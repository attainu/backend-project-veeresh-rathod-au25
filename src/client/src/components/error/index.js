import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles, Typography, Button } from '@material-ui/core';
import { Body } from '../common';

const useStyles = makeStyles(() => ({
  container: {
    margin: "10em auto",
    maxWidth: "32em"
  }
}));

const Error = (props) => {
  const classes = useStyles();

  return (
    <Body>
      <section className={classes.container}>
        <Typography variant="h2" color="secondary">
          Sorry...
        </Typography>
        <br />
        <Typography variant="h5">
          {props.message}
        </Typography>
        <br />
        <Button href="/app" size="large" variant="outlined" color="primary">
          Log in
        </Button>
      </section>
    </Body>
  );
};

Error.propTypes = {
  message: PropTypes.string
};

export default Error;
