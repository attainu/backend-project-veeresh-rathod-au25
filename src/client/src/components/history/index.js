import React from 'react';
import PropTypes from 'prop-types';
import times from 'lodash/times';
import map from 'lodash/map';
import { makeStyles, Typography, List } from '@material-ui/core';
import { blue, lime, orange, red, yellow, indigo, cyan, amber, grey } from '@material-ui/core/colors';

import HistoryItem from './history-item';
import { Body } from '../common';

const allColors = [];
times(5, () => allColors.push(
  blue, lime, orange, red, yellow, indigo, cyan, amber
));

const useColors = makeStyles(() => map(allColors, (c) => ({
  color: "white",
  backgroundColor: c[500]
})));

const useStyles = makeStyles(() => ({
  container: {
    marginTop: "1em",
    textAlign: "left"
  },
  title: {
    padding: "8px 16px"
  },
  noData: {
    color: grey[500],
    padding: "16px"
  }
}));

const History = ({ users }) => {
  const colors = useColors();
  const classes = useStyles();

  return (
    <Body isLoggedIn={true}>
      <section className={classes.container}>
        <Typography variant="h4" color="primary" className={classes.title}>
          User History
        </Typography>
        {!users.length ? (
          <Typography variant="h5" className={classes.noData}>No history data</Typography>
        ) : <List>
            {users.map((user, i) => (
              <HistoryItem
                key={user.username}
                color={colors[i]}
                {...user} />)
            )}
          </List>}
      </section>
    </Body>
  );
};

History.propTypes = {
  users: PropTypes.array
};

export default History;