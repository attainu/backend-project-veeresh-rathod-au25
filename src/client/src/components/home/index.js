import React, { useState } from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import { makeStyles, Modal, Paper, Typography } from '@material-ui/core';

import ActiveUsers from "./active-users";
import PopoverContent from "./popover-content";
import { Body } from "../common";
const MAX = 5;

const useStyles = makeStyles(() => ({
  container: {
    height: "85%",
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  },
  title: {
    display: "flex",
    flex: 0.4,
    alignItems: "center"
  },
  main: {
    display: "flex",
    flex: 0.2,
    alignItems: "center"
  },
  footer: {
    display: "flex",
    flex: 0.4,
    alignItems: "flex-end"
  },
  modalWindow: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  modalCtr: {
    width: "50%",
    overflow: "auto",
    maxHeight: "75%"
  }
}));

const Home = ({ activeUsers, loggedInUser }) => {
  const classes = useStyles();
  const [showDialog, setDialog] = useState(false);

  const isAdmin = get(loggedInUser, "isAdmin", false);
  const handleClose = () => setDialog(false);
  const shownUsers = activeUsers.slice(0, MAX);
  const hiddenUsers = activeUsers.slice(MAX);

  return (
    <Body isLoggedIn={true} admin={isAdmin}>
      <section className={classes.container}>
        <header className={classes.title}>
          <Typography variant="h4">
            Welcome {isAdmin ? "Admin!" : "User!"}
          </Typography>
        </header>
        <main className={classes.main}>
          <ActiveUsers
            users={shownUsers}
            hiddenCount={hiddenUsers.length}
            max={MAX}
            handleMore={() => setDialog(true)} />
          <Modal
            open={showDialog}
            onClose={handleClose}
            className={classes.modalWindow}
          >
            <Paper className={classes.modalCtr}>
              <PopoverContent users={hiddenUsers} />
            </Paper>
          </Modal>
        </main>
        <footer className={classes.footer}>
          <Typography variant="caption">
            Created by: Amruth Skanda, 2020
        </Typography>
        </footer>
      </section>
    </Body>
  );
};

Home.propTypes = {
  activeUsers: PropTypes.array,
  loggedInUser: PropTypes.object
};

export default Home;