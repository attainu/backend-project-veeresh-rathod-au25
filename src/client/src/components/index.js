export { default as Login } from "./login";
export { default as Home } from "./home";
export { default as History } from "./history";
export { default as Registration } from "./registration";
export { default as Error } from "./error";
