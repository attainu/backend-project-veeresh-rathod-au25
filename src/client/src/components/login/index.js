import React from 'react';
import PropTypes from 'prop-types';
import { Paper, makeStyles } from '@material-ui/core';
import Alert from '@material-ui/lab/Alert';

import { Body } from "../common";
import LoginForm from './login-form';
import { login } from '../../actions';

const useStyles = makeStyles(() => ({
  loginCtr: {
    margin: "8em auto",
    maxWidth: "32em"
  }
}));

const Login = ({
  loggingIn,
  loginFailure,
  signingUp,
  signUpSuccess,
  loggingOut,
  logoutSuccess
}) => {
  const classes = useStyles();
  const showError = !loggingIn && loginFailure;
  const registerSuccess = !signingUp && signUpSuccess;
  const logoutSuccessAlert = !loggingOut && logoutSuccess;
  const handleLogin = ({ username, password }) => {
    login(username.value, password.value);
  };

  return (
    <Body>
      <section className={classes.loginCtr}>
        {registerSuccess && <Alert severity="success">Registration successfull! Please login below</Alert>}
        {logoutSuccessAlert && <Alert severity="success">Logout successful! Visit again!</Alert>}
        {showError && <Alert severity="error">Invalid credientials. Please try again!</Alert>}
        <br />
        <Paper elevation={2}>
          <LoginForm
            onSubmit={handleLogin}
            isSubmitting={loggingIn} />
        </Paper>
      </section>
    </Body>
  );
};

Login.propTypes = {
  loggingIn: PropTypes.bool,
  loginFailure: PropTypes.bool,
  signingUp: PropTypes.bool,
  signUpSuccess: PropTypes.bool,
  loggingOut: PropTypes.bool,
  logoutSuccess: PropTypes.bool
};

export default Login;