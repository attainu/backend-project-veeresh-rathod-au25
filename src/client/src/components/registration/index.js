import React from 'react';
import PropTypes from 'prop-types';
import Alert from '@material-ui/lab/Alert';
import { Paper, makeStyles } from '@material-ui/core';

import { Body } from "../common";
import RegistrationForm from './registration-form';
import { signUp } from '../../actions';

const useStyles = makeStyles(() => ({
  signUpCtr: {
    margin: "10em auto",
    maxWidth: "32em"
  }
}));

const Registration = ({ signingUp, signUpFailure }) => {
  const classes = useStyles();
  const showError = !signingUp && signUpFailure;
  const handleRegistration = ({ username, password, admin }) => {
    signUp(username.value, password.value, admin.checked);
  };

  return (
    <Body>
      <section className={classes.signUpCtr}>
        {showError && <Alert severity="error">{signUpFailure}</Alert>}
        <br />
        <Paper elevation={2}>
          <RegistrationForm
            isSubmitting={signingUp}
            onSubmit={handleRegistration} />
        </Paper>
      </section>
    </Body>
  );
};

Registration.propTypes = {
  signingUp: PropTypes.bool,
  signUpFailure: PropTypes.string
};

export default Registration;