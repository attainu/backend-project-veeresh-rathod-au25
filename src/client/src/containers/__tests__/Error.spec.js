import React from 'react';
import ErrorContainer from "../Error";
import { Error } from "../../components";
import { shallow } from "enzyme";

describe("CONTAINER: src/containers/Error.js", () => {
  const render = (props) => shallow(<ErrorContainer {...props} />);

  it("should match snapshot", () => {
    expect(render()).toMatchSnapshot();
  });

  it("should render Error component with proper meesage prop", () => {
    const defaultMsg = "Invalid session. Please login again!";
    expect(render().find(Error).prop("message")).toBe(defaultMsg);
  })
});