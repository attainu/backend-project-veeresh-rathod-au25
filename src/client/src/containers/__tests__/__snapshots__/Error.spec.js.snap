// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTAINER: src/containers/Error.js should match snapshot 1`] = `ShallowWrapper {}`;
